subprojects/workers/src/main/java/org/gradle/workers/WorkerExecutor.java

```diff
@@ -20,26 +20,38 @@
 import org.gradle.api.Incubating;
 
 /**
- * Worker Executor.
+ * Allows work to be submitted for asynchronous execution.  Work should be submitted with a {@link Runnable} class
+ * representing the implementation of the unit of work and an action to configure the unit of work (via {@link WorkerConfiguration}).
+ *
+ * <pre>
+ *      workerExecutor.submit(RunnableWorkImpl.class) { WorkerConfiguration conf ->
+ *          // Set the isolation mode for the worker
+ *          conf.isolationMode = IsolationMode.NONE
+ *
+ *          // Set up the constructor parameters for the unit of work
+ *          conf.params = [ "foo", file('bar') ]
+ *      }
+ * </pre>
  *
  * @since 3.5
  */
 @Incubating
 public interface WorkerExecutor {
     /**
-     * Submits a piece of work to be executed.
+     * Submits a piece of work to be executed asynchronously.
      *
      * Execution of the work may begin immediately.
      *
-     * Forked work will execute in an idle daemon that meets the requirements set on this builder.  If no
-     * idle daemons are available, a new daemon will be started.  Any errors will be thrown from  {@link #await()}.
-     *
-     * In the event that an error is thrown while submitting work, all uncompleted work will be canceled.
+     * Work configured with {@link IsolationMode#PROCESS} will execute in an idle daemon that meets the requirements set
+     * in the {@link WorkerConfiguration}.  If no idle daemons are available, a new daemon will be started.  Any errors
+     * will be thrown from {@link #await()} or from the surrounding task action if {@link #await()} is not used.
      */
     void submit(Class<? extends Runnable> actionClass, Action<? super WorkerConfiguration> configAction);
 
     /**
-     * Blocks until all work associated with the current build operation is complete.
+     * Blocks until all work associated with the current build operation is complete.  Note that when using this method inside
+     * a task action, it will block completion of the task action until all submitted work is complete.  This means that other
+     * tasks from the same project cannot be run in parallel while the task action is still executing.
```

Comment:
This seems limiting to some degree.  Is there a way to submit three units of work in parallel (pre-reqs) and then wait for all three to finish before submitting a 4th unit of work, all in one task, and NOT block any other parallel tasks?